### ISSUE TITLE
Always search for devices

### ISSUE THREAD

**Issue by 749801324** [Cierpliwy/SensorTag#5]
When my service starts.
It's always in the state of searching sensors.
Buttons are not available
Do I want to update the version?

**Comment by shamprasadrh** [comment]
I am also facing the same issue.

**Comment by JeffShomali** [comment]
#metoo

**Comment by hugAdmin** [comment]
The App is looking for a peripheral with a name of 'Sensor Tag'

Ln 150: Saga.js

if (scannedDevice != null && scannedDevice.localName === "SensorTag") {
          emit([error, scannedDevice]);
        }

Remove the check for that in the if statement and you'll see the next available peripheral appearing in the log. Hit 'Forget' to cycle through the list.

**Comment by acteq** [comment]
correct. i found the same problem yesterday, and change the code from 
if (scannedDevice != null && scannedDevice.localName === "SensorTag") {
emit([error, scannedDevice]);
}
to
if (scannedDevice != null ) {
emit([error, scannedDevice]);
}

**Comment by brukyoung** [comment]
okay removed the condition but still can't discover my device does this scan phones as well

**Comment by gism** [comment]
Thanks for the code!

I was facing same issue. Using Android I was not able to connect to my device.
In following:
```
function* scan(manager: BleManager): Generator<*, *, *> {
  if (Platform.OS === 'android' && Platform.Version >= 23) {
    yield put(log('Scanning: Checking permissions...'));
    const enabled = yield call(
      PermissionsAndroid.check,
      PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION,
    );
    if (!enabled) {
      yield put(log('Scanning: Permissions disabled, showing...'));
      const granted = yield call(
        PermissionsAndroid.request,
        PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION,
      );
      if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
        yield put(log('Scanning: Permissions not granted, aborting...'));
        // TODO: Show error message?
        return;
      }
    }
  }
```
Replace `PermissionsAndroid.PERMISSIONS.ACCESS_COARSE_LOCATION` for `PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION`
